### PR TITLE
MS20893 [CreateApp] Ignore children of mismatching entityHostTypes

### PR DESCRIPTION
**Pull request by thoys**
skip child entities of mismatching entityHostType when copy/delete/duplication of entities

This PR should fix the following tickets:

- https://highfidelity.manuscript.com/f/cases/20893/When-deleting-a-zone-and-then-undoing-the-delete-it-creates-a-second-persistent-preview-shape-which-is-visible-to-everyone
- https://highfidelity.manuscript.com/f/cases/20892/New-Zone-going-Opaque

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16903/HighFidelity-Beta-PR14806-132d293.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16903/HighFidelity-Beta-PR14806-132d293.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16903/HighFidelity-Beta-Interface-PR14806-132d293.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16903/HighFidelity-Beta-PR14806-132d293.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16903/HighFidelity-Beta-Interface-PR14806-132d293.exe

**Comment by QliCknox** [comment]
Currently Testing

**Comment by QliCknox** [comment]
Pass

Zones no longer duplicate into movable entities with create closed

File: scripts/system/edit.js [hunk 1]
```diff
@@ -1694,21 +1706,24 @@ function deleteSelectedEntities() {
...
-                });
-                deletedIDs.push(entityID);
-                Entities.deleteEntity(entityID);
+            if (initialProperties.locked ||
+                (initialProperties.avatarEntity && initialProperties.owningAvatarID !== MyAvatar.sessionUUID)) {
+                continue;
             }
+            var children = Entities.getChildrenIDs(entityID);
+            var childList = [];
+            recursiveDelete(children, childList, deletedIDs, entityHostTypes[i].entityHostType);
```
Comment (huffman) [inline scripts/system/edit.js]: This kind of changes the behavior, right? Previously we would delete all avatar or domain entities, now we only delete one or the other, dependent on the root entity that we want to delete. Do we want that?

File: scripts/system/edit.js [hunk 2]
```diff
@@ -1608,30 +1608,42 @@ function sortSelectedEntities(selected) {
     return sortedEntities;
 }
 
-function recursiveDelete(entities, childrenList, deletedIDs) {
+function recursiveDelete(entities, childrenList, deletedIDs, entityHostType) {
+    var wantDebug = false;
     var entitiesLength = entities.length;
-    for (var i = 0; i < entitiesLength; i++) {
+    var initialPropertySets = Entities.getMultipleEntityProperties(entities);
+    var entityHostTypes = Entities.getMultipleEntityProperties(entities, 'entityHostType');
```
[the remark this answers is not in the document]
Comment (thoys) [reply scripts/system/edit.js]: that is correct, and we do need all these properties for the undo stack.

**Pull request merged**